arwen/elrondapi/elrondei.go

```diff
@@ -393,6 +424,85 @@ func transferValue(context unsafe.Pointer, destOffset int32, valueOffset int32,
 	return 0
 }
 
+//export createAsyncCall
+func createAsyncCall(context unsafe.Pointer,
+	asyncContextIdentifier int32,
+	identifierLength int32,
+	destOffset int32,
+	valueOffset int32,
+	dataOffset int32,
+	length int32,
+	successOffset int32,
+	successLength int32,
+	errorOffset int32,
+	errorLength int32,
+	gasPercentage int32,
+) {
+	runtime := arwen.GetRuntimeContext(context)
+
+	acIdentifier, err := runtime.MemLoad(asyncContextIdentifier, identifierLength)
+	if arwen.WithFault(err, context, runtime.ElrondAPIErrorShouldFailExecution()) {
+		return
+	}
+
+	calledSCAddress, err := runtime.MemLoad(destOffset, arwen.AddressLen)
+	if arwen.WithFault(err, context, runtime.ElrondAPIErrorShouldFailExecution()) {
+		return
+	}
+
+	value, err := runtime.MemLoad(valueOffset, arwen.BalanceLen)
+	if arwen.WithFault(err, context, runtime.ElrondAPIErrorShouldFailExecution()) {
+		return
+	}
+
+	data, err := runtime.MemLoad(dataOffset, length)
+	if arwen.WithFault(err, context, runtime.ElrondAPIErrorShouldFailExecution()) {
+		return
+	}
+
+	successFunc, err := runtime.MemLoad(successOffset, successLength)
+	if arwen.WithFault(err, context, runtime.ElrondAPIErrorShouldFailExecution()) {
+		return
+	}
+
+	errorFunc, err := runtime.MemLoad(errorOffset, errorLength)
+	if arwen.WithFault(err, context, runtime.ElrondAPIErrorShouldFailExecution()) {
+		return
+	}
+
+	err = runtime.AddAsyncContextCall(acIdentifier, &vmcommon.AsyncGeneratedCall{
+		Destination: calledSCAddress,
+		Data: data,
+		ValueBytes: value,
+		SuccessCallback: string(successFunc),
+		ErrorCallback: string(errorFunc),
+		GasPercentage: gasPercentage,
+	})
+	if arwen.WithFault(err, context, runtime.ElrondAPIErrorShouldFailExecution()) {
+		return
+	}
+
+}
+
+//export createAsyncContextCallback
+func createAsyncContextCallback(context unsafe.Pointer, asyncContextIdentifier int32, identifierLength int32, callback int32) int32 {
```

Comment:
Yes, will fix it and add a test case for it so it will actually be used. This function adds a callback for an entire context of async calls, example - all ACs in: `"my_first_vacation"`

Reply:
So it's like `Promise.finally()` in JS?

Reply:
Done, fixed the code with the correct functionality